### PR TITLE
Gas Consumption Supply Crud

### PR DESCRIPTION
**Pull request by shadowalker001**
Gas Consumption Supply Crud

### CODE REVIEW

File: database/migrations/2024_09_22_192745_create_daily_volumes_table.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+     */
+    public function up(): void
+    {
+        Schema::create('daily_volumes', function (Blueprint $table) {
+            $table->id();
+            $table->bigInteger("customer_id")->comment('customer id');
+            $table->bigInteger("customer_site_id")->comment('customer site id');
+            $table->float('volume')->comment('volume in Scf');
+            $table->float('rate')->comment('rate in NGN/Scf, should be picked from current rate in settings');
+            $table->float('amount')->comment('amount from (volume * rate) in NGN');
```
Comment (kingsonly) [inline database/migrations/2024_09_22_192745_create_daily_volumes_table.php]: remove the amount and rate

File: app/Services/DailyVolumeService.php [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+        $query = DailyVolume::query();
+
+        // Apply dynamic filters
+        foreach ($filters as $key => $value) {
+            // Handle date range filters for created_at and updated_at
+            if ($key === 'created_at_from') {
+                $query->whereDate('created_at', '>=', $value);
+            } elseif ($key === 'created_at_to') {
+                $query->whereDate('created_at', '<=', $value);
+            } elseif ($key === 'updated_at_from') {
```
Comment (kingsonly) [inline app/Services/DailyVolumeService.php]: this would have been better as a switch case

File: app/Services/DailyVolumeService.php [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+    }
+
+    /**
+     * Get all customer daily volumes.
+     *
+     * @return \Illuminate\Database\Eloquent\Collection|DailyVolume[]
+     */
+    public function getAll()
+    {
+        return DailyVolume::all();
```
Comment (kingsonly) [inline app/Services/DailyVolumeService.php]: this should be paginated

File: app/Services/DailyVolumeService.php [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+     * @throws \Throwable
+     */
+    public function create(array $data): DailyVolume
+    {
+        Log::info('Starting daily volume creation process', ['data' => $data]);
+
+        try {
+            // Check if form_field_answers is provided in JSON format
+            if (isset($data['form_field_answers'])) {
+                // Decode the JSON data
```
Comment (kingsonly) [inline app/Services/DailyVolumeService.php]: add transaction to this

File: app/Http/Controllers/DailyVolumeController.php [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+    public function index(Request $request)
+    {
+        try {
+            // Get all request parameters for filtering
+            $filters = $request->all();
+
+            // Pass the filters to the service for query building
+            $dailyVolumes = $this->dailyVolumeService->getAllWithFilters($filters);
+
+            return DailyVolumeResource::collection($dailyVolumes)->additional([
```
Comment (kingsonly) [inline app/Http/Controllers/DailyVolumeController.php]: this should be paginated

Comment (kingsonly) [reply app/Http/Controllers/DailyVolumeController.php]: the default should be query for the current month

File: app/Http/Controllers/DailyVolumeController.php [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+    {
+        try {
+            $dailyVolume = $this->dailyVolumeService->getById($id);
+            $dailyVolume->delete();
+
+            return response()->json([
+                'status' => 'success',
+                'message' => 'Daily Volume record deleted successfully'
+            ], 204);
+        } catch (\Illuminate\Database\Eloquent\ModelNotFoundException $e) {
```
Comment (kingsonly) [inline app/Http/Controllers/DailyVolumeController.php:242]: delete follows the same approach as always, but we need to think about the approach for delete

**Review by kingsonly — CHANGES_REQUESTED** [review]

**Review by Skillz-systems — APPROVED** [review]

**Pull request merged**